src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -0,0 +1,217 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.flowpowered.math.vector.Vector3d;
+import com.flowpowered.math.vector.Vector3f;
+import com.flowpowered.math.vector.Vector3i;
+import org.apache.commons.lang3.Validate;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.Queries;
+import org.spongepowered.api.data.type.DyeColor;
+
+import java.util.Objects;
+
+public final class Color implements DataSerializable {
+
+    private static final int MASK = 0xff;
+
+    public static final Color BLACK = ofRgb(0x000000);
+
+    public static final Color GRAY = ofRgb(0x808080);
+
+    public static final Color WHITE = ofRgb(0xFFFFFF);
+
+    public static final Color BLUE = ofRgb(0x0000FF);
+
+    public static final Color GREEN = ofRgb(0x008000);
+
+    public static final Color LIME = ofRgb(0x00FF00);
+
+    public static final Color RED = ofRgb(0xFF0000);
+
+    public static final Color YELLOW = ofRgb(0xFFFF00);
+
+    public static final Color MAGENTA = ofRgb(0xFF00FF);
+
+    public static final Color PURPLE = ofRgb(0xAA00FF);
+
+    public static final Color DARK_CYAN = ofRgb(0x008B8B);
+
+    public static final Color DARK_GREEN = ofRgb(0x006400);
+
+    public static final Color DARK_MAGENTA = ofRgb(0x8B008B);
+
+    public static final Color NAVY = ofRgb(0x000080);
+
+    public static final Color UNKNOWN = ofRgb(0xFF00AA);
+
+    public static Color ofRgb(int hex) {
+        return ofRgb(hex >> 16 & MASK, hex >> 8 & MASK, hex & MASK);
+    }
+
+    public static Color ofRgb(int red, int green, int blue) {
+        return new Color(red, green, blue);
+    }
+
+    public static Color of(java.awt.Color color) {
+        return new Color(color.getRed(), color.getGreen(), color.getBlue());
+    }
+
+    public static Color of(Vector3i vector3i) {
+        return new Color(vector3i.getX(), vector3i.getY(), vector3i.getZ());
+    }
+
+    public static Color of(Vector3f vector3f) {
+        return new Color(vector3f.getFloorX(), vector3f.getFloorY(), vector3f.getFloorZ());
+    }
+
+    public static Color of(Vector3d vector3d) {
+        return new Color(vector3d.getFloorX(), vector3d.getFloorY(), vector3d.getFloorZ());
+    }
+
+    public static Color mixDyeColors(DyeColor... colors) {
+        Validate.noNullElements(colors, "No nulls allowed!");
+        final Color[] actualColors = new Color[colors.length];
+        for (int i = 0; i < colors.length; i++) {
+            actualColors[i] = colors[i].getColor();
```

Comment:
Floating point colours are always normalized, 0 is min, 1 is max. The byte range makes no sense here. Scale the values and round instead of flooring.

> On Nov 18, 2015, at 2:38 AM, Minecrell <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/api/util/Color.java:
> 
> > +        return new Color(vector3i.getX(), vector3i.getY(), vector3i.getZ());
> > +    }
> > +
> > +    public static Color of(Vector3f vector3f) {
> > +        return new Color(vector3f.getFloorX(), vector3f.getFloorY(), vector3f.getFloorZ());
> > +    }
> > +
> > +    public static Color of(Vector3d vector3d) {
> > +        return new Color(vector3d.getFloorX(), vector3d.getFloorY(), vector3d.getFloorZ());
> > +    }
> > +
> > +    public static Color mixDyeColors(DyeColor... colors) {
> > +        Validate.noNullElements(colors, "No nulls allowed!");
> > +        final Color[] actualColors = new Color[colors.length];
> > +        for (int i = 0; i < colors.length; i++) {
> > +            actualColors[i] = colors[i].getColor();
> It just seems a bit inconsistent to use Validate and then again preconditions, but whatever. You can keep it if you want
> 
> —
> Reply to this email directly or view it on GitHub.
>